exercises/practice/grade-school/test_grade_school.c

```diff
@@ -13,10 +14,11 @@ void tearDown(void)
 
 static void populate_roster(roster_t * expected)
 {
-   for (size_t i = 0; i < expected->count; ++i)
-      TEST_ASSERT_TRUE(add_student
-                       (expected->students[i].name,
-                        expected->students[i].grade));
+   for (size_t i = 0; i < expected->count; ++i) {
+      char name_buffer[MAX_NAME_LENGTH + 1];
```

Comment:
`MAX_NAME_LENGTH` is part of the implementation, technically students are free to change it or remove it completely... this would then break.

Reply:
If it's changed but not removed, this will change accordingly because it's using the macro from the implementation's header.

Reply:
> Your call, I can go either way. If we decide to keep the existing behavior, we should remote `MAX_NAME_LENGTH` since it is inconsistent with using statically allocated strings.

Go with this new approach, if we get feedback that possible solutions are too complex then we can adjust again a needed.